### PR TITLE
Fixing GDB error if ' exists in path

### PR DESCRIPTION
**Pull request by WardenGnaw**
Addresses issue for the C++ Extension

https://github.com/Microsoft/vscode-cpptools/issues/1554

### CODE REVIEW

File: src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs [hunk 1]
```diff
@@ -815,7 +815,7 @@ private void AddExecutablePathCommand(IList<LaunchCommand> commands)
                 throw new LaunchErrorException(message);
             };
 
-            commands.Add(new LaunchCommand("-file-exec-and-symbols " + exe, description, ignoreFailures: false, failureHandler: failureHandler));
+            commands.Add(new LaunchCommand("-file-exec-and-symbols \"" + exe + "\"", description, ignoreFailures: false, failureHandler: failureHandler));
```
Comment (pieandcakes) [inline src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs]: Please add a comment for why we're adding quotes.

**Comment by pieandcakes** [comment]
Otherwise LGTM. :shipit:

**Comment by gregg-miskelly** [comment]
>             if (path.IndexOf(' ') != -1)

Your fix should be here instead

---
Refers to: src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs:1324 in b673208. [](commit_id = b673208fee7bf2d263f0216098ffcbc32890e600, deletion_comment = False)

**Comment by WardenGnaw** [comment]
Would I also need to add it in https://github.com/Microsoft/MIEngine/blob/d7e8da83d1fa1e719c949bf1f192147e09752160/src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs#L1335

It currently works in VsCode:
```
1: (1453) <-1012-symbol-list-lines E:\\weirdfolder'\\main.cpp
1: (1454) ->1012^done,lines=[{pc="0x00401460",line="4"},{pc="0x00401471",line="4"},{pc="0x00401476",line="5"},{pc="0x0040149b",line="6"},{pc="0x004014a0",line="7"},{pc="0x004014a8",line="0"},{pc="0x004014ba",line="7"},{pc="0x004014c0",line="7"},{pc="0x004014cf",line="0"},{pc="0x004014e5",line="7"},{pc="0x004014e8",line="7"},{pc="0x004014ee",line="7"},{pc="0x00401504",line="0"}]
```

File: src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs [hunk 2]
```diff
@@ -1321,7 +1321,7 @@ internal string EnsureProperPathSeparators(string path)
                 path = path.Replace(@"\", @"\\");
             }
 
-            if (path.IndexOf(' ') != -1)
+            if (path.IndexOf(' ') != -1 || path.IndexOf('\'') != -1)
```
Comment (gregg-miskelly) [inline src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs]: nit: use `IndexOfAny`. Also, is it worth checking other special characters? Or do we just want to fix these when people complain about them?

Comment (pieandcakes) [reply src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs]: I think we can take the tact of updating it when there is reports of other characters that it doesn't like. This is the first time I've seen this come up.

**Pull request merged**